### PR TITLE
Migrate API tests to Kotlin

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR migrates tests from `com.google.common.flogger` package to Kotlin.

Please note, these core tests will be migrated in a separate PR:

1. `LogContextTest`
2. `AbstractLoggerTest`
3. `FluentLogger2Test`

### CODE REVIEW

**Comment by yevhenii-nadtochii** [comment]
@alexander-yevsyukov PTAL

File: flogger/flogger-api-tests/src/test/kotlin/com/google/common/flogger/given/BackendTestEnv.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+package com.google.common.flogger.given
+
+/**
+ * Contains test environment for flogger API tests.
+ */
+
+/**
+ * Creates a new [Iterator] over the given [values].
+ */
+internal fun <T> iterate(vararg values: T): Iterator<T> = listOf(*values).iterator()
```
Comment (alexander-yevsyukov) [inline flogger/flogger-api-tests/src/test/kotlin/com/google/common/flogger/given/BackendTestEnv.kt:36]: Just curious, isn't there a standard Kotlin way to do such a thing?

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM, with one minor request and a question.

File: flogger/flogger-api-tests/src/test/kotlin/com/google/common/flogger/given/BackendTestEnv.kt (hunk 1 above)
Comment (yevhenii-nadtochii) [reply flogger/flogger-api-tests/src/test/kotlin/com/google/common/flogger/given/BackendTestEnv.kt:36]: Unfortunately, no. I've checked.

**Pull request merged**